### PR TITLE
C4C 285 Add/Edit Intentions

### PR DESCRIPTION
**Pull request by hackeret**

### CODE REVIEW

File: C4CCoreApp/C4CCoreUITests/InventionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (rekowp17) [inline C4CCoreApp/C4CCoreUITests/InventionsTests.cs]: This file should be renamed to IntentionsTests :)

File: C4CCoreApp/C4CCoreApp/Controllers/IntentionController.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            Intention intention = new Intention();
+            intention = Intention.GetIntention(conn, new Guid(id));
+
+            return View("~/Views/Intention/Edit.cshtml", intention);
+        }
+
+        [AcceptVerbs("POST")]
+        [HttpPost]
+        [Route("Edit")]
+        public ActionResult update(Intention intention)
```
Comment (codyhaws) [inline C4CCoreApp/C4CCoreApp/Controllers/IntentionController.cs]: Minor style edit: Capitalize the method name (C# convention)

File: C4CCoreApp/C4CCoreApp/Models/AccountViewModels/DeleteUserViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace C4CCoreApp.Models.AccountViewModels
+{
+    /// <summary>
+    /// The model used for deleting users.
+    /// </summary>
+    public class DeleteUserViewModel
+    {
+
```
Comment (codyhaws) [inline C4CCoreApp/C4CCoreApp/Models/AccountViewModels/DeleteUserViewModel.cs]: What is this file necessary for, if it's empty?

File: C4CCoreApp/C4CCoreApp/Views/Intention/Intention.cshtml [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+        if (id) {
+            $("#editDiv").load("@Html.Raw("Edit")", id);
+        } else {
+            $("#editDiv").load("@Html.Raw("Add")", id);
+        }
+    }
+</script>
+
+<style>
```
Comment (codyhaws) [inline C4CCoreApp/C4CCoreApp/Views/Intention/Intention.cshtml]: We have CSS files in the wwwroot/css folder, you can move styles in there if you want... but it's not an issue.

Comment (hackeret) [reply C4CCoreApp/C4CCoreApp/Views/Intention/Intention.cshtml]: I tried putting them in the wwwroot/css folder initially but it didn't render for some reason
